morello/src/search.rs

```diff
@@ -64,9 +64,9 @@ enum WorkingPartialImpl<Tgt: Target> {
     Sat,
 }
 
-#[derive(Debug, Clone)] // TODO: Remove this Clone derive if not needed
+#[derive(Debug)]
 struct ImplReducer {
-    results: SmallVec<[(ActionIdx, Cost); 1]>,
+    results: BTreeSet<(Cost, ActionIdx)>,
     top_k: usize, // TODO: Shared between ImplReducers. Pull out?
```

Comment:
@samkaufman  I think `top_k` means that in each `ImplReducer`, the invariant of keeping `<= top_k` elements should hold (in the `insert` function).  I feel it is weird to get affected from the outside of `ImplReducer`, once it's constructed.  Even though other `ImplReducer`'s hold the same value, I feel pulling it out does not make sense?  What do you think?

```suggestion
    top_k: usize,
```